provider/azure/azure.go

```diff
@@ -65,10 +91,18 @@ type AzureProvider struct {
 // NewAzureProvider creates a new Azure provider.
 //
 // Returns the provider or an error if a provider could not be created.
-func NewAzureProvider(configFile string, domainFilter endpoint.DomainFilter, zoneNameFilter endpoint.DomainFilter, zoneIDFilter provider.ZoneIDFilter, resourceGroup string, userAssignedIdentityClientID string, dryRun bool) (*AzureProvider, error) {
+func NewAzureProvider(
+	configFile string,
+	domainFilter endpoint.DomainFilter,
+	zoneNameFilter endpoint.DomainFilter,
+	zoneIDFilter provider.ZoneIDFilter,
+	resourceGroup string,
+	userAssignedIdentityClientID string,
+	dryRun bool,
+) (*AzureProvider, error) {
 	cfg, err := getConfig(configFile, resourceGroup, userAssignedIdentityClientID)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read Azure config file '%s': %v", configFile, err)
+		return nil, fmt.Errorf("failed to read Azure config file %q: %v", configFile, err)
```

Comment:
nit: prefer `%w` to `%v` when wrapping errors.

Reply:
Done in [`9b24560` (#3285)](https://github.com/kubernetes-sigs/external-dns/pull/3285/commits/9b245607b738ea247c7cb938fa2700b508060e9b)